### PR TITLE
new_savanna_dry_dirt

### PR DESCRIPTION
**Pull request by TumeniNodes**
adds new savanna dirt "dry_dirt" as requested in https://github.com/minetest/minetest_game/issues/2405
default_dry_dirt

this was achieved by using 0.5 gaussian blur x2 to the original 'dirt' texture, then recolouring, and then going through and editing various pixels by hand, and testing with each alteration to decrease linear tiling as best as possible.

Everything can be tweaked at request, this is just to move on the issue
Screenshot from 2019-07-14 10-14-21
Screenshot from 2019-07-14 10-14-35

screenshot_20190714_100458
screenshot_20190714_100512

### CODE REVIEW

**Comment by Napiophelios** [comment]
I think the texture is decent look for dry dirt;
but dirt is gonna be dry in savanna no matter how deep you go? 
or just what's exposed to air or attached to grass?

**Comment by TumeniNodes** [comment]
> I think the texture is decent look for dry dirt;
> but dirt is gonna be dry in savanna no matter how deep you go?
> or just what's exposed to air or attached to grass?

That's a good question. I'm not sure how deep one would need to dig in the Savanna to reach richer soil. Or if that is even something to think about in a game.
Either way, that's something I guess paramat can look into and set up ingame.

File: mods/default/nodes.lua [hunk 1]
```diff
@@ -453,14 +453,21 @@ minetest.register_node("default:dirt_with_grass_footsteps", {
 	}),
 })
 
-minetest.register_node("default:dirt_with_dry_grass", {
-	description = "Dirt with Dry Grass",
+minetest.register_node("default:dry_dirt", {
+	description = "Savanna Dirt",
```
Comment (paramat) [inline mods/default/nodes.lua]: I think
'Dry Dirt'
to make it more flexible and not linked to a biome?

File: mods/default/nodes.lua [hunk 2]
```diff
@@ -453,14 +453,21 @@ minetest.register_node("default:dirt_with_grass_footsteps", {
...
+	tiles = {"default_dry_dirt.png"},
+	groups = {crumbly = 3, soil = 1},
+	sounds = default.node_sound_dirt_defaults(),
+})
+
+minetest.register_node("default:dry_dirt_with_dry_grass", {
+	description = "Savanna Dirt with Dry Grass",
```
Comment (paramat) [inline mods/default/nodes.lua]: 'Dry Dirt with ...'
?

File: mods/default/nodes.lua [hunk 3]
```diff
@@ -453,14 +453,21 @@ minetest.register_node("default:dirt_with_grass_footsteps", {
...
 	tiles = {"default_dry_grass.png",
-		"default_dirt.png",
-		{name = "default_dirt.png^default_dry_grass_side.png",
+		"default_dry_dirt.png",
+		{name = "default_dry_dirt.png^default_dry_grass_side.png",
 			tileable_vertical = false}},
 	groups = {crumbly = 3, soil = 1, spreading_dirt_type = 1},
-	drop = 'default:dirt',
+	drop = 'default:dry_dirt',
```
Comment (paramat) [inline mods/default/nodes.lua]: Sorry to be perfectionist but we are changing all ' to " in node names

File: mods/default/mapgen.lua [hunk 4]
```diff
@@ -576,8 +597,28 @@ function default.register_ores()
 		},
 		-- Only where default:dirt is present as surface material
 		biomes = {"taiga", "snowy_grassland", "grassland", "coniferous_forest",
-			"deciduous_forest", "deciduous_forest_shore", "savanna", "savanna_shore",
-			"rainforest", "rainforest_swamp", "floatland_coniferous_forest"}
+			"deciduous_forest", "deciduous_forest_shore", "rainforest", "rainforest_swamp", "floatland_coniferous_forest"}
```
Comment (paramat) [inline mods/default/mapgen.lua]: Please wrap at 90 columns (with tab size 4)

File: mods/default/mapgen.lua [hunk 5]
```diff
@@ -113,6 +113,27 @@ function default.register_mgv6_ores()
 		},
 	})
 
+	-- Dry Dirt
+
+	minetest.register_ore({
```
Comment (paramat) [inline mods/default/mapgen.lua]: Savanna biome doesn't occur in mgv6 so this ore is not needed.

File: mods/default/nodes.lua [hunk 6]
```diff
@@ -48,7 +48,7 @@ Soft / Non-Stone
 default:dirt
 default:dirt_with_grass
 default:dirt_with_grass_footsteps
-default:dirt_with_dry_grass
```
Comment (paramat) [inline mods/default/nodes.lua]: Also need to add
default:dry_dirt

**Comment by paramat** [comment]
Thanks was hoping you'd work on a texture.
PR looks quite good already and i'll test later.

**Comment by paramat** [comment]
As a suggestion for the texture:

I like the 'texture' of the texture, but would prefer a more interesting colour than light brown. Savanna dirt is sometimes noticeably orange-red hued, or a light sandy colour, or a light greyish-sandy colour, all would be more interesting than light brown.
The MTG savanna is dry season, yellow grass, so dirt would be baked by the sun and possibly very light in colour, almost as light as the grass.

As an example of a red soil:

depositphotos_202374592-stock-photo-red-sand-track-stretching-wilderness

Sandy:



Later i'll probably make patches of savanna bare dirt, so it will have this sandy appearence.

**Comment by paramat** [comment]
> but dirt is gonna be dry in savanna no matter how deep you go?

Yes, for simplicity.

File: mods/default/nodes.lua (hunk 2 above)
Comment (TumeniNodes) [reply mods/default/nodes.lua]: done

File: mods/default/nodes.lua (hunk 3 above)
Comment (TumeniNodes) [reply mods/default/nodes.lua]: done

File: mods/default/mapgen.lua (hunk 4 above)
Comment (TumeniNodes) [reply mods/default/mapgen.lua]: done (I hope)

File: mods/default/nodes.lua (hunk 6 above)
Comment (TumeniNodes) [reply mods/default/nodes.lua]: done

**Comment by TumeniNodes** [comment]
The new colour. The old one had a slight hint of red but, obviously it was not detectible.
I used the colours from the image you posted above to see if this colour appeals more.
default_dry_dirt

<details><summary>Images (click to uncollapse)</summary>
<p>

screenshot_20190714_191629
screenshot_20190714_191640
screenshot_20190714_191711

<p>
</details>

File: mods/default/mapgen.lua (hunk 5 above)
Comment (TumeniNodes) [reply mods/default/mapgen.lua]: done

**Comment by paramat** [comment]
Thanks, looks good to me.
We could always tune it later after i add bare dirt patches to savanna and we see what that looks like.

**Comment by paramat** [comment]
There's now a conflict due to new grass texture being merged, needs rebase unfortunately.

**Comment by DS-Minetest** [comment]
Uh, there will be many unknown nodes.
And adding something like dry dirt is a very big change.
Newly generated parts of the world will look very differently.
Does the new dry grass spread? (And does dirt_with_snow still spread?)

**Comment by paramat** [comment]
> Uh, there will be many unknown nodes.

Hmm why? Have i missed something? =)

We always make biome changes, these always make newly generated world different, and better. Dry dirt isn't that much of a change as it's mostly covered in dry grass.

Dry grass will no longer spread, see the issue for why.

File: mods/default/mapgen.lua [hunk 7]
```diff
@@ -576,8 +576,29 @@ function default.register_ores()
 		},
 		-- Only where default:dirt is present as surface material
 		biomes = {"taiga", "snowy_grassland", "grassland", "coniferous_forest",
-			"deciduous_forest", "deciduous_forest_shore", "savanna", "savanna_shore",
-			"rainforest", "rainforest_swamp", "floatland_coniferous_forest"}
+				"deciduous_forest", "deciduous_forest_shore", "rainforest",
+				"rainforest_swamp", "floatland_coniferous_forest"}
+	})
+
+	minetest.register_ore({
```
Comment (paramat) [inline mods/default/mapgen.lua]: This was good and thorough of you to add this, but i prefer to not add another blob ore.
Dirt blobs are just added to biomes that have normal dirt, savanna simply no longer has normal dirt, just like tundra has permafrost dirt.

**Comment by paramat** [comment]
> Uh, there will be many unknown nodes.

Ugh oops, of course, we can't remove the 'dirt_with_dry_grass' node.
Thanks.

File: mods/default/nodes.lua [hunk 8]
```diff
@@ -48,7 +48,8 @@ Soft / Non-Stone
...
```
Comment (paramat) [inline mods/default/nodes.lua]: Node must be kept to avoid unknown nodes.

File: mods/default/nodes.lua [hunk 9]
```diff
@@ -447,20 +448,27 @@ minetest.register_node("default:dirt_with_grass_footsteps", {
...
 			tileable_vertical = false}},
 	groups = {crumbly = 3, soil = 1, not_in_creative_inventory = 1},
-	drop = 'default:dirt',
+	drop = "default:dirt",
 	sounds = default.node_sound_dirt_defaults({
 		footstep = {name = "default_grass_footstep", gain = 0.25},
 	}),
 })
 
-minetest.register_node("default:dirt_with_dry_grass", {
```
Comment (paramat) [inline mods/default/nodes.lua]: Node must be kept to avoid unknown nodes.

File: mods/default/mapgen.lua (hunk 7 above)
Comment (TumeniNodes) [reply mods/default/mapgen.lua]: done, and rebased the branch successfully (I think)

File: mods/default/nodes.lua (hunk 8 above)
Comment (TumeniNodes) [reply mods/default/nodes.lua]: done

File: mods/default/nodes.lua (hunk 9 above)
Comment (TumeniNodes) [reply mods/default/nodes.lua]: done

**Comment by TumeniNodes** [comment]
rebasing blows... I will just redo the PR. Should have it up in a couple hours.

**Comment by TumeniNodes** [comment]
new PR https://github.com/minetest/minetest_game/pull/2412

**Comment by TumeniNodes** [comment]
this PR is now fubar

**Pull request closed without merging**